### PR TITLE
feat(Reddit): add privacy mode

### PR DESCRIPTION
**Pull request by realtofuine**
## Description 

add privacy mode to reddit rich presence

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image

image

</details>

### CODE REVIEW

File: websites/R/Reddit/presence.ts [hunk 1]
```diff
@@ -40,7 +40,8 @@ const startTimestamp = Math.floor(Date.now() / 1000),
 presence.on("UpdateData", async () => {
 	setPresence();
 	const newLang = await presence.getSetting<string>("lang").catch(() => "en"),
-		buttons = await presence.getSetting<boolean>("buttons");
+		buttons = await presence.getSetting<boolean>("buttons"),
+		privacy = await presence.getSetting<boolean>("privacy");
```
Comment (EncryptedDev) [inline websites/R/Reddit/presence.ts:44]: Use [Promise.all()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all). 
Extensive reference: https://github.com/PreMiD/Presences/blob/5ccc89aca1c18bb5e8de37dca18f3e659e134b7b/websites/A/Apple/presence.ts#L96-L113

File: websites/R/Reddit/presence.ts [hunk 2]
```diff
@@ -54,102 +55,140 @@ presence.on("UpdateData", async () => {
 		{ pathname } = window.location;
 	if (oldReddit) {
 		subReddit = document.querySelector(".redditname")
-			? `r/${document.querySelector(".redditname").textContent}`
+			? `${
+					!privacy
+						? `r/${document.querySelector(".redditname").textContent}` // if privacy mode is disabled
+						: "In a subreddit" // if privacy mode is enabled
```
Comment (EncryptedDev) [inline websites/R/Reddit/presence.ts:61]: This should be localised

File: websites/R/Reddit/presence.ts [hunk 3]
```diff
@@ -54,102 +55,140 @@ presence.on("UpdateData", async () => {
...
+				];
+			} else {
+				// if privacy mode is enabled
+				presenceData.details = "Reading a post";
+				presenceData.state = subReddit;
+			}
+		} else if (pathname.startsWith("/user/")) {
+			if (!privacy) {
+				// if privacy mode is disabled
+				username = document.querySelector(".titlebox > h1").textContent;
```
Comment (EncryptedDev) [inline websites/R/Reddit/presence.ts:84]:
```suggestion
				presenceData.state = document.querySelector(".titlebox > h1").textContent;
```

File: websites/R/Reddit/presence.ts [hunk 4]
```diff
@@ -54,102 +55,140 @@ presence.on("UpdateData", async () => {
...
+				presenceData.details = strings.profile;
+				presenceData.state = username;
```
Comment (EncryptedDev) [inline websites/R/Reddit/presence.ts:86]:
```suggestion
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Hello 👋

Please remove the comments, they explain exactly what a boolean expression is and are not needed.

**Comment by EncryptedDev** [comment]
Format with `yarn format`

File: websites/R/Reddit/presence.ts [hunk 5]
```diff
@@ -54,102 +58,136 @@ presence.on("UpdateData", async () => {
...
+			presenceData.details = strings.browsing;
 			presenceData.state = !sub2 ? "Home" : sub2.textContent;
 		} else {
-			presenceData.details = (await strings).browsing;
+			presenceData.details = strings.browsing;
 			presenceData.state = sub.textContent;
 		}
+	} else {
+		const sub = document.querySelector("span._1GieMuLljOrqnVpRAwz7VP");
+		if (sub === null) {
```
Comment (EncryptedDev) [inline websites/R/Reddit/presence.ts:179]:
```suggestion
		if (!sub) {
```

File: websites/R/Reddit/presence.ts [hunk 6]
```diff
@@ -171,6 +209,6 @@ presence.on("UpdateData", async () => {
 			});
 		}
 	}
-	if (!buttons) delete presenceData.buttons;
+	if (!buttons || privacy) delete presenceData.buttons; // delete button if show button is disabled or privacy mode is enabled
 	presence.setActivity(presenceData, true);
```
Comment (EncryptedDev) [inline websites/R/Reddit/presence.ts:213]:
```suggestion
	presence.setActivity(presenceData);
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**